src/System.Private.CoreLib/shared/System/Collections/ArrayList.cs

```diff
@@ -909,7 +909,7 @@ public override int IndexOf(object? value, int startIndex)
 
             public override int IndexOf(object? value, int startIndex, int count)
             {
-                if (startIndex < 0 || startIndex > Count) throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
+                if ((uint)startIndex > (uint)Count) throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
```

Comment:
Technically a breaking change since the `get_Count` virtual method is now always called, even if _startIndex_ is negative. Honestly I don't care, but wanted to point it out since it's an example of something the JIT wouldn't be able to optimize automatically since it'd likely pessimistically assume the `get_Count` virtual method might have side effects.

(This same comment applies to a few other places in this file.)